src/concepts/nodes.md

```diff
@@ -0,0 +1,72 @@
+# Nodes
+
+## What are Nodes?
+
+---
+
+**Nodes** are simply **actions** or **conditions**. By connecting Nodes together complex action sequences based on certain conditions are created.
+
+Nodes are visually represented by Caido as draggable blocks, colorized by category. They utilize an input/output model that can be used to send data altered by one Node to a subsequent Node.
+
+<img alt="Convert workflow" src="/_images/workflow_convert_basic.png" center/>
+
+## Connecting Nodes
+
+---
+
+A `Connection` is visually represented by the gray line between nodes and determine the order of execution.
+
+- Workflows use a top-down heirachical structure (_the Node at the very top represents the beginning of the flow and the Node at the bottom represents the end of the flow_).
+
+<img alt="Workflow Connections" src="/_images/node_connect.png" center/>
+
+1. The down arrow within a circle icon represents a Node's `socket`.
+2. Click and drag a bottom socket to the top socket of the next Node in the flow to create a `Connection`.
+
+## Node Alias
+
+---
+
+A Node's `Alias` is an arbitrarily set identifier used to uniquely reference the associated Node within a Workflow.
+
+> _NOTE: This differs from a Node's Name which simply serves cosmetic purposes to assist in the visual representation._
+
+## Node Input Types: Constant Value & Reference Value
+
+---
+
+### 1. Constant Value Type
+
+`Constant Values` - the input used by the Node's execution will be the **_supplied value_**.
+
+- To use this input type, manually enter the data to be used in the `Data` field under `Inputs` with the `Use reference` checkbox **_deselected_**.
+
+<img alt="Base64 Node" src="/_images/const_value_node.png" center/>
+
+### 2. Reference Value Type
+
+`Reference Values` - the input used by the Node's execution will be the **_output of a previous Node_**.
+
+- To use this input type, the content of the `Data` field under `Inputs` with the `Use reference` checkbox **_selected_** should be formatted using the following syntax:
+
+```
+$[node_alias].[property_alias]
+```
+
+<img alt="Workflow Node Inputs" src="/_images/reference_value_node.png" center/>
+
+_Example (pictured above):_
+
+- _The value_ `$start.data` _is the output of the_ **Start** _Node being taken as input by the_ **Base64 Encode** _Node_.
+- _The output of the_ **Base64 Encode** _Node will be referenced by the_ **End** _Node as_ `$base64_encode.data`.
+
+## Node Categories
+
+---
+
+- **Encoding/Decoding/Hashing/Dehashing Nodes** (Blue) - Base64/Hex/HTML/JWT/MD5/SHA1/SHA2/URL
```

Comment:
I prefer we something along:
- **Start/Stop Nodes** (Yellow)
- **Control Nodes** (Green)
  Like `If/Else`, this allows you to control the execution flow (ref execution flow) 
- **Code Nodes** (Red)
  Like `Shell`, `JS`, those allow you to... 
- **All Other Nodes** (Blue)

Also the list here is only for convert workflows, it is different right now for active/passive. We should mention that each type has different nodes based on its capabilities and that we are expanding them (https://github.com/caido/caido/issues?q=is%3Aopen+is%3Aissue+label%3A%22topic%3A+workflow%22).